### PR TITLE
feat[152]: use unified types for STT input and output

### PR DESCRIPTION
**Pull request by John-A-J**

### CODE REVIEW

File: src/Playthrough.ts [hunk 1]
```diff
@@ -449,7 +450,9 @@ class Playthrough extends EventEmitter<PlaythroughEvents> {
     }
   }
 
-  private onSpeechRecognitionResult = (event: any): void => {
+  private onSpeechRecognitionResult = (
+    event: SpeechRecognitionResponse,
```
Comment (bensalilijames) [inline src/Playthrough.ts:454]: also adjust `PlaythroughEvents` type above

File: src/types.ts [hunk 2]
```diff
@@ -257,6 +257,18 @@ export type ConfirmStartEvent = ConfirmEvent<ConfirmStartEventPayload>;
 export type ConfirmTapEvent = ConfirmEvent<ConfirmTapEventPayload>;
 
 export type SpeechRecognitionStartEvent = {
-  service: string;
-  serviceOptions: any;
+  service: "unified" | "unified:google" | "unified:aws" | "unified:deepgram";
+  sampleRate?: number;
+  languageCode?: string;
+  encoding?: string;
+  customServiceParameters?: object;
```
Comment (bensalilijames) [inline src/types.ts:264]: the `object` type also includes arrays, let's be more specific in what we expect

```suggestion
  customServiceParameters?: Record<string, unknown>;
```

File: test/mic-check.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      languageCode: "en-US",
+      returnRaw: false,
+      customServiceParameters: {}
+    };
+    const charismaUrl = "http://localhost:5050";
+    //
+
+    setGlobalBaseUrl(charismaUrl);
+
+    console.log();
```
Comment (bensalilijames) [inline test/mic-check.html:41]: I know we're the only ones that will end up using this test file, but would be good to tidy up empty comment/console.log

File: test/mic-check.html [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        const div = document.createElement("div");
+        div.classList.add("message", "character");
+        div.innerHTML = `<b>${message.message.character?.name || "???"}</b>: ${
+          message.message.text
+        }`;
+        messagesDiv.appendChild(div);
+      });
+
+      playthrough.on('speech-recognition-result', (speechResult) => {
+        if(speechResult.isFinal) {
```
Comment (bensalilijames) [inline test/mic-check.html:73]: the lack of a space after `if` suggests `prettier` hasn't auto-formatted this doc, worth checking VS Code settings to see if it's running on `.html` files

File: test/mic-check.html [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+      microphone.on("timeout", () => console.log("Timed out!"));
+      microphone.on("recognise", (text) => {
+        console.log("recognise", text);
+        conversation.reply({ text });
+      });
+      microphone.on("recognise-interim", (text) => {
+        console.log("recognise-interim", text);
+        microphone.resetTimeout(timeout);
+      });
```
Comment (bensalilijames) [inline test/mic-check.html:105]: let's remove all references to `microphone` (this test file can be for the speech recognition proxy only - `microphone` is for the built-in browser speech recognition)

**Review by bensalilijames — CHANGES_REQUESTED** [review]

File: test/mic-check.html [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+          } from "../dist/index.js";
+
+    import {
+      setGlobalBaseUrl,
+    } from "../dist/api.js";
+
+    // TEST PARAMETERS
+    const storyId = 2;
+    const version = -1;
+      const userToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYXV0aGVudGljYXRlZCIsInVzZXJfaWQiOiI3MDA5ZDNmMS1iYTE0LTRhMDEtYTJhOC1hNjA5ZTdlN2NkOGYiLCJpYXQiOjE2OTE2NzIwMDIsImV4cCI6MTY5NDI2NDAwMiwiYXVkIjoicG9zdGdyYXBoaWxlIiwiaXNzIjoicG9zdGdyYXBoaWxlIn0.DMu6QJjOu7yISGQ-ajM0sua9VON9zvxNyEdrKB4uq2I";
```
Comment (bensalilijames) [inline test/mic-check.html:29]: not a massive deal to have this here as nothing sensitive inside it but might want to consider redacting, or maybe turning all these config things into env vars: https://en.parceljs.org/env.html

File: test/mic-check.html [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  <button onclick="recordingStart()">⏺</button>
+  <button onclick="recordingStop()">⏹</button>
+</body>
+<script type="module">
+  import {
+    Playthrough,
+    Speaker,
+            createPlaythroughToken,
+            createConversation,
+          } from "../dist/index.js";
```
Comment (bensalilijames) [inline test/mic-check.html:20]: investigate some formatting funkiness here

**Review by bensalilijames — CHANGES_REQUESTED** [review]

File: test/mic-check.html (hunk 6 above)
Comment (John-A-J) [reply test/mic-check.html]: I don't think it's worth the effort, adds a layer of complexity and values will need to be manually input every time anyway

**Review by bensalilijames — APPROVED** [review]

**Pull request merged**